lalrpop-util/Cargo.toml

```diff
@@ -17,7 +17,7 @@ regex-automata = { workspace = true, optional = true }
 lexer = ["regex-automata/std", "std"]
 unicode = ["regex-automata?/unicode"]
 std = []
-default = ["std"]
+default = ["std", "unicode"]
```

Comment:
Should `unicode` be default of `lalrpop-util` too? Because `lalrpop/unicode` now requires `lalrpop-util/unicode`, wouldn't this be enabled by default from the default `lalrpop/unicode` feature?

If we are going to add `unicode` here, we also have to add `no-default-feature` flag to `lalrpop`'s `lalrpop-util` dependency. Otherwise `lalrpop-util/unicode` will never be disabled from `lalrpop` crate.

Reply:
Ah, good point. Maybe the other change of this PR, which is that `larlpop/unicode` implies `lalrpop-util/unicode`, is sufficient alone?

Reply:
Thinking through this, I'm wondering about the following situation:

Library crate `foo` is a user of lalrpop.  They have `lalrpop` in their build dependencies, and `lalrpop-util` in their dependencies, using default features for both.  Another crate `bar` uses `foo` as a dependency.

I believe the situation here if we have `lalrpop/unicode` imply `lalrpop-util/unicode`, but don't set `unicode` as a default feature in `lalrpop-util` is that due to feature unification, `foo` will have the `unicode` feature set on their local builds, but when `bar` builds, they end up doing a build that never brings in `lalrpop`, just `lalrpop-util`, since `lalrpop was only a build time dependency, and so `lalrpop-util` doesn't inherit the `unicode` feature in `bar`'s build, which could of course cause problems if `foo` relies on unicode (possibly unknowningly).

That seems to me to be an argument in favor of adding `no-default-feature` to `lalrpop` as per the second paragraph of @youknowone's comment, but keeping the default on `lalrpop-unicode` to avoid the feature unification issue I just described?

This is all theory based on how I think feature unification works.  If it would be helpful, I can set up a collection of crates with the above structure and verify that that's how it works in practice.

Reply:
Adding no-default-feature makes sense to me for that cases

Reply:
Pushed in line with that.